src/libsync/owncloudpropagator.h

```diff
@@ -227,8 +240,20 @@ private slots:
         return next->scheduleSelfOrChild();
     }
 
+    void terminateJobs() {
+        // Send abort to the subjobs and emit abort finished signal
+        abortJobs();
```

Comment:
The idea here was to force the abort. Thus if after 5 seconds they did not finish, you kill the job just by sending abort jobs again, please have a look on chunkingng implementation